readthedocs/vcs_support/backends/git.py

```diff
@@ -115,12 +121,19 @@ def get_remote_fetch_refspec(self):
             # denoting that it's not a branch/tag that really exists.
             # Because we don't know if it originates from the default branch or some
             # other tagged release, we will fetch the exact commit it points to.
-            if self.version_machine and self.verbose_name == "stable":
+            if self.version_machine and self.verbose_name == STABLE_VERBOSE_NAME:
                 if self.version_identifier:
                     return f"{self.version_identifier}"
                 log.error("'stable' version without a commit hash.")
                 return None
-            return f"refs/tags/{self.verbose_name}:refs/tags/{self.verbose_name}"
+
+            tag_name = self.verbose_name
+            # For a machine created "latest" tag, the name of the tag is set
+            # in the `Version.identifier` field, note that it isn't a commit
+            # hash, but the name of the tag.
+            if self.version_machine and self.verbose_name == LATEST_VERBOSE_NAME:
+                tag_name = self.version_identifier
+            return f"refs/tags/{tag_name}:refs/tags/{tag_name}"
```

Comment:
The docstring of this function explains all the cases. Does it need to be updated with this change?

Reply:
Went ahead and re-wrote the docstring https://github.com/readthedocs/readthedocs.org/pull/10738/commits/fcc7b4b010c3bf473638a65074d67a617cb37e75